src/App.vue

```diff
@@ -148,6 +148,19 @@
         </v-container>
       </v-content>
       <snackbar/>
+      <v-dialog v-model="createTeamDialog" width="500">
+          <v-card>
+            <v-toolbar dark>
+                <v-btn icon dark @click="createTeam">
+                <v-icon>close</v-icon>
+                </v-btn>
+                <v-toolbar-title class="white--text">Create a new team</v-toolbar-title>
+            </v-toolbar>
+            <v-card-text>
+              <CreateTeam />
```

Comment:
All the things with vuex state to manage `createTeamDialog` is too complex. You should just be able to manage local state in this App component. Add an event listener to `CreateTeam` which emits event when form is submitted and then hide the dialog from current component

```html
<CreateTeam @close="dialog = false"/>
```